cluster/pre-existing/util.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A library of helper functions for a pre-existing Kubernetes cluster
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../..
+source "${KUBE_ROOT}/cluster/common.sh"
+source "${KUBE_ROOT}/hack/lib/util.sh"
+
+function detect-project() {
+  if [[ -z "${MASTER_IP-}" ]]; then
+    echo "Set 'MASTER_IP' to the instance assigned to be the Kubernetes master" 1>&2
+    exit 1
+  fi
+
+  if [[ -z "${PROJECT-}" ]]; then
```

Comment:
Since normally ```detect-project``` sets the value of ```PROJECT```, I was imitating that here.  But, adding ```PROJECT="${PROJECT:-kubemark}"``` to ```config-default.sh``` for the pre-existing provider will accomplish the same thing.